test/e2e/tcp/connectivity.go

```diff
@@ -167,5 +169,17 @@ func createPods(p *ConnectivityTestParams) (*framework.NetworkPod, *framework.Ne
 
 	framework.Logf("Connector pod has IP: %s", connectorPod.Pod.Status.PodIP)
 
+	// In Globalnet deployments, when backend pods finish their execution, kubeproxy-iptables driver tries
+	// to delete the iptables-chain associated with the service (even when the service is present) as there are
+	// no active backend pods. Since the iptables-chain is also referenced by Globalnet Ingress rules, the chain
+	// cannot be deleted (kubeproxy errors out and continues to retry) until Globalnet removes the reference.
+	// Globalnet removes the reference only when the service itself is deleted. Until Globalnet is enhanced [*]
+	// to remove this dependency with iptables-chain, lets delete the service after the listener Pod is terminated.
+	// [*] https://github.com/submariner-io/submariner/issues/1166
+	if p.ToEndpointType == GlobalIP {
```

Comment:
Shouldn't we just go ahead and delete them for ServiceIP ones too?

Reply:
We don't have any issue with Vanilla Submariner (where ServiceIP) are used. Only globalnet e2e tests use endpointType as tcp.GlobalIP